deepchem/models/tests/test_kerasmodel.py

```diff
@@ -319,15 +319,19 @@ def test_wandblogger():
        tf.keras.layers.Dense(1)])
   model = dc.models.KerasModel(
       keras_model, dc.models.losses.L2Loss(), wandb_logger=wandblogger)
-  vc = dc.models.ValidationCallback(valid_dataset, 1, [metric])
-  model.fit(train_dataset, nb_epoch=10, callbacks=[vc])
+  vc_train = dc.models.ValidationCallback(train_dataset, 1, [metric])
+  vc_valid = dc.models.ValidationCallback(valid_dataset, 1, [metric])
+  model.fit(train_dataset, nb_epoch=10, callbacks=[vc_train, vc_valid])
+  # call model.fit again to test multiple fit() calls
+  model.fit(train_dataset, nb_epoch=10, callbacks=[vc_train, vc_valid])
+  wandblogger.finish()
```

Comment:
Will users need to call `finish()`, or are you only doing this so you can examine the log?  If someone just trains a model normally, is it ok for them to skip calling it?

Reply:
Normally, it's best practice to call `finish()` for wandb since that ends the run and shuts off the wandb backend process. Without calling finish, nothing will crash but the wandb process will still be running until the next run or once the notebook/script is closed/shutoff (it might also prevent files such as checkpoints from being uploaded).

Currently `wandblogger.finish()` also saves the run history data for testing usage. I can move the saving run history data into its own function if you'd like? But I think we would still need to call finish() at some point.

If you don't think users should call `wandblogger.finish()`, is there a better place where you think I should add finish() to end the run? fit() was the only place that I could think of originally.

Reply:
There isn't really any good place for it.  The typical usage is 1) create the model, 2) train it by calling `fit()` some number of times, 3) evaluate it either with `evaluate()` or in some other manner.  If the script exits without having called `finish()`, will the process get shut down and all data saved correctly?  If so, I think that's ok.

Reply:
`wandblogger.finish()` makes a call to [wandb.finish()](https://docs.wandb.ai/ref/python/finish) at the end. However as it says on the doc `wandb.finish()` will also be called automatically when a script exits. I'm not too certain about the behaviour for notebooks since you could still run cells after the run finishes and I'm not sure if shutting down a notebook is equivalent to exiting a script. I always explicitly call finish() just to make sure all data is saved and uploaded.

So I suppose it would still be okay if users don't call `wandblogger.finish()` but I think it makes sense to mention it in the docs for best practices.

Reply:
Is there something you can call that ensures all data is saved and uploaded, but doesn't shut down the process?

Reply:
I went back and checked with the team on this: There would be no risk of data not being saved or uploaded if `finish()` is not called. `finish()` just ends the wandb python process (in a small minority of scenarios this might be important). A user could choose to not call `finish()` and they would be perfectly fine. Apologies for the confusion earlier.

Most users do call `wandb.finish()` and in other integrations with libraries such as Huggingface or Pytorch Lightning users might explicitly call `wandb.finish()` if there's no `finish()` wrapper function inside the Wandb logger class. They might even save additional data after training the model and then once they decide they are done they would call `wandb.finish()`. We also emphasize `wandb.finish()` a lot in our docs too.

Since our WandbLogger is importing wandb inside the class itself (importing wandb in the logger class is a standard we are gradually moving towards), we don't require users to import wandb at the beginning of their code and so there's no good way to call `wandb.finish()` since wandb is not always imported. Therefore, we would just expose a `WandbLogger.finish()` function that calls `wandb.finish()` so that users can shut down the wandb process.
Our opinion is that we think that `wandblogger.finish()` should be kept just so that it's clear that users can access something that calls wandb.finish() if they need to.

Reply:
That sounds fine to me.